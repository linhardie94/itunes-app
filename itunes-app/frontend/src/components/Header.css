/*this is my css for my header component*/
.Header {
    overflow: hidden;
    background-color: #f1f1f1;
}

.Header p {
    float: left;
    color: black;
    text-align: center;
    padding: 10px;
    text-decoration: none;
    font-size: 18px;
    line-height: 15px;
    border-radius: 4px;
}

.Header p.logo {
    font-size: 25px;
    font-weight: bold;
    color: #5c5239;
}

.Header p:hover {
    background-color: #ddd;
    color: #5d6770;
}

.Header p.active {
    background-color: #917c46;
    color: white;
}

.HeaderRight {
    float: right;
}

@media screen and (max-width: 500px) {
    .Header p {
        float: none;
        display: block;
        text-align: left;
    }

    .HeaderRight {
        float: none;
    }
}