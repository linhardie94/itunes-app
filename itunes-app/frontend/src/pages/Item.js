//single item from itunes
import React from 'react';
import styled from 'styled-components';
import Palette from './Palette';

//card css
const Card = styled.div`
	height: 200px;
  width: 150px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: center;
  white-space: nowrap;
	margin: 10px 5px;
	padding: 5px;
	background-color: ${Palette('Grey', 600)};
	color: ${Palette('White')};
	:hover {
		box-shadow: 0 0 0.75rem ${Palette('Secondary', 'dark')};
	}
`;

//card text css
const CardText = styled.div`
	overflow: hidden;
	text-overflow: ellipsis;
`;

//creating the cards for each item returned from itunes
const Item = ({ kind, artistName, trackName, artworkUrl100 }) => (
	<Card>
		<div>{kind}</div>
		<CardText>{artistName}</CardText>
		<CardText>{trackName}</CardText>
		<div>
			<img src={artworkUrl100} />
		</div>
	</Card>
);

export default Item;
