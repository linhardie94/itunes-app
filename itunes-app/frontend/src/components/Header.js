//this is my header component
import React, {useEffect, useState} from 'react';
import { Link, useLocation } from 'react-router-dom';
import "./Header.css"

const Header = () => {
    const [activeTab, setActiveTab] = useState("Home");

    //setting active tab to show when you're on the page
    const location = useLocation();
    useEffect(() => {
        if(location.pathname === "/") {
            setActiveTab("Home")
        } else if(location.pathname === "/add") {
            setActiveTab("AddFavourite")
        } else if(location.pathname === "/search") {
            setActiveTab("Search")
        }
    }, [location])

    //setting up the header component to respond on click and show active tab
  return (
    <div className='Header'>
        <p className='Logo'>Your iTunes Favourites List</p>
        <div className='HeaderRight'>
                <Link to="/">
                    <p  
                        className={`${activeTab === "Home" ? "active" : ""}`} 
                        onClick={() => setActiveTab("Home")}
                    >
                        Home
                    </p>
                </Link>
                <Link to="/add">
                    <p 
                        className={`${activeTab === "AddFavourite" ? "active" : ""}`} 
                        onClick={() => setActiveTab("AddFavourite")}
                    >
                        Add favourite
                    </p>
                </Link>
                <Link to="/search">
                    <p 
                        className={`${activeTab === "Search" ? "active" : ""}`} 
                        onClick={() => setActiveTab("Search")}
                    >
                        Search
                    </p>
                </Link>
            </div>
    </div>
  )
}

export default Header