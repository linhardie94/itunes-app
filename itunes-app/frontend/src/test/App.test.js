import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header.js';
import Item from '../pages/Item.js';
import { render, screen, act } from '@testing-library/react';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Item />).toJSON();
  expect(tree).toMatchSnapshot();
});

/*
test('renders learn weather app', () => {
  render(<App />);
  const p = screen.getByText(/Your iTunes Favourites List/i);
  expect(p).toBeInTheDocument();
});


it('renders correctly when the p is found', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toBeInTheDocument('Your iTunes Favourites List');
});
*/

it('renders correctly when there are no items', () => {
  const leaf = renderer.create(<Header />).toJSON();
  expect(leaf).toBeInDocument('Add');
});