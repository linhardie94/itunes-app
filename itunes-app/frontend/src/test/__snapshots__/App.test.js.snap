// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="sc-bcXHqe dqFPfI"
>
  <div />
  <div
    className="sc-gswNZR gjWgKi"
  />
  <div
    className="sc-gswNZR gjWgKi"
  />
  <div>
    <img />
  </div>
</div>
`;
